### ISSUE TITLE
Syntax for the job configuration and query 

### ISSUE THREAD

**Issue by srao712** [addthis/hydra#195]
Hi,
Where can I get the exact syntax/format of all the sections and parameters in the job configuration file and the queries. The reference document provides all the parameters, but only a few examples are provided, not the exhaustive syntax. Same is the case for queries.

**Comment by tipat300** [comment]
Gift

**Comment by tea-dragon** [comment]
You mean the configuration language syntax? It accepts hocon (a more permissive variant of json).

**Comment by abramsm** [comment]
I think the question is more for a comprehensive list of hydra syntax.
This is an area where documentation could be vastly improved.  The current
docs are available here:

http://oss-docs.addthiscode.net/hydra/latest/user-guide/

But we would love contributions to help improve.

matt

**Comment by mxk1235** [comment]
I am struggling to come up with a source definition which reads json records from a gzipped file, e.g. event.2016-02-01.json.gz.  

can you point at an existing example or the relevant section of the documentation (which seems outdated)?

i am also getting the following error while running hydra-local as in the tutorial (v5.5.30-SNAPSHOT)  `Error saving job: no eligible hosts for new task`.  The message is quite cryptic.  Is it not supposed to work in local mode?

**Comment by tea-dragon** [comment]
Just to quickly address that second error: probably your hard drive is nearly full and any/all local minions you have think they're too loaded to take new tasks. Last time it happened to me, I simply cleaned up old heap dumps, but I can find you the settings to ignore that if needed.

**Comment by tea-dragon** [comment]
As to the source issue, the gz part is automatically handled by every source that comes to mind. So the real question is: how is your json file formatted?

- a big array of strings?
- newline separated kv maps?
- an array of kv maps?

... And so on. I guess most importantly: " is the entire file a single json element? "

**Comment by mxk1235** [comment]
hmmm...  well, this is local, so i don't think disk space is a problem.

```mxk@mxk-mbp ~/workspace/hydra/hydra-local/streams (master) $ df
Filesystem    512-blocks      Used Available Capacity   iused    ifree %iused  Mounted on
/dev/disk1     974700800 805679336 168509464    83% 100773915 21063683   83%   /```

the json file simply has a json object on each line.  i guess i just need "type":"json", but i can't find an example that uses that, and playing around with "columns" and "tokens" from this example doesn't seem to be doing anything, although i might be blocked by the host issue.

**Comment by tea-dragon** [comment]
Looks like you're at 83% disk usage. That's probably enough to trip minion
warnings by default :/

As for the json source, what you'd want is `format: json`. Eg.

```
source.mesh2 {
    mesh.files = "data.json.gz"
    format = json
}
```

**Comment by tea-dragon** [comment]
https://github.com/addthis/hydra/blob/master/hydra-main/src/main/java/com/addthis/hydra/job/spawn/balancer/SpawnBalancerConfig.java#L75

There are the system properties that decide how much free disk space
minions need to accept new tasks and fun existing tasks respectively (80%
and 90% by default). We'll change the local stack settings to be much more
forgiving in the future, but you can override those system properties as a
quick hack. Or well, I guess you're building it anyway, lol.

**Comment by tea-dragon** [comment]
Actually went ahead and made the config change for local stack, so you
could just pull and you should be okay.
